### PR TITLE
Edit profile (Details)

### LINKED ISSUE #90: Create UI part of user profile
Create only ui part of profile.

This must be new page "/profile". 

Something like this....
image

Or ...

image

### PR DESCRIPTION
**Pull request by Vova-Romaniuk**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/users/maryanchuk04/projects/3)

## Code reviewers

- [ ] @maryanchuk04 
- [ ] @Serhiy-dotcom 

### Second Level Review

- [ ] @Varvara-Maria 

## Summary of issue

Edit profile info UI Part

## Summary of change

Create new page edit profile
made changes in  previosly created files

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: StudCity.Frontend/src/pages/RegistrationComplete/steps/UploadAvatar.jsx [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
 	const [currentPage, setCurrentPage] = useState("choose-img");
 	const [imgAfterCrop, setImgAfterCrop] = useState("");
-
+	
 	const onImageSelected = (selectImg) => {
 		setImage(selectImg);
 		setCurrentPage("crop-img");
 	}
-
+	useEffect(() => {
```
Comment (maria2908) [inline StudCity.Frontend/src/pages/RegistrationComplete/steps/UploadAvatar.jsx]: I think we don`t need that console.log

**Review by maria2908 — CHANGES_REQUESTED** [review]

File: StudCity.Frontend/src/UI/CustomDialog/index.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import Button from "../Button";
+function CustomDialog({ className = "", children, close }) {
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/UI/CustomDialog/index.jsx]: rename close prop into handleClose as this more epropriate naming convention to use

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+import Button from "../../../UI/Button";
+import TextField from "../../../UI/fields/TextField";
+import SelectField from "../../../UI/fields/SelectField";
+import ValidateTextField from "../../../UI/fields/ValidateTextField";
+import UploadAvatar from "../../RegistrationComplete/steps/UploadAvatar";
+import { 
+	SELECT_BACKGROUND_URLS,
+	DEFAULT_BACKGROUND_URL,
+ } from "../../../utils/constants";
+//import UploadImageInput from "../../../UI/fields/UploadImageInput";
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: remove comment please

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	const spanWithInputsBlocks = "ml-16 text-left text-base";
+	const textFieldStyle = "w-60 h-10";
+	const [avatar, setAvatar] = useState("");
+	const [backgroundImage, setBackgroundImage] = useState("");
+	const [showDialogForAvatar, setShowDialogForAvatar] = useState(false);
+	const [showDialogForBackground, setShowDialogForBackground] = useState(false);
+	return(
+		<div className="w-full h-fit flex flex-col">
+			{showDialogForAvatar 
+				?
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: you can remove ternary and use absolut AND instead

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+					<UploadAvatar 
+						className={"w-[800px] h-[600px]"} 
+						setAvatar={setAvatar}
+					/>
+				</CustomDialog> 
+				: 
+				<></> 
+			}
+			{showDialogForBackground 
+				?
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: here as well

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+				<></> 
+			}
+			{showDialogForBackground 
+				? 
+				<CustomDialog close={setShowDialogForBackground}>
+					<div className="w-[1000px] h-[600px] flex ">
+						<div className="w-4/5 h-[90%] overflow-scroll grid grid-cols-2 m-auto gap-10">
+							{
+								SELECT_BACKGROUND_URLS.map(( element, index ) => (
+									<div className="rounded-tl-[60px] overflow-hidden w-full  h-56" key = {index}>
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: don't use index as key, better use element instead if it's unique

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+			<div className="w-10/12 h-full flex flex-col mx-auto pb-11">
+				<div className="h-24 w-full  border-b-2 border-[#D1D7D4] flex items-center">
+					<h4 className="text-3xl font-medium" >Settings</h4>
+				</div>
+				<div className="h-5/6 flex w-full">
+					<div className="w-3/12 h-full "></div>
+					<div className="w-9/12 h-full flex flex-col">
+						<div className="h-52 w-full overflow-hidden rounded-tl-[60px] mt-6 shadow-form relative">
+							<img 
+								src={backgroundImage === "" ? DEFAULT_BACKGROUND_URL : backgroundImage}
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: better use just backgroundImage and set DEFAULT_BACKGROUND_URL directly into state as default

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+								className="absolute bottom-3 right-3 w-12 h-12 bg-transparent border-2 border-white "
+								onClick={() => setShowDialogForBackground(true)}
+							>
+								<i className="fa-regular fa-camera"></i>
+							</Button>
+						</div>
+						<div className="h-32 w-full -mt-10 flex justify-between">
+							<div className="h-fit w-fit relative mx-auto">
+								<Avatar 
+									src={avatar === "" ? "/images/defaultAvatar.png" : avatar}
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: same here

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+							<div className={`${inputBlock}`}>
+								<span className={`${spanWithInputsBlocks}`}>Phone number</span>
+								<ValidateTextField
+									className={`${textFieldStyle}`}
+									withErrorMessage={false}	
+									placeholder={"Input your phone"}
+								/>
+							</div>
+							<div className={`${inputBlock}`}>
+								<span className={`${spanWithInputsBlocks}`}>Birthday</span>
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: no need to use tildas here, use className={spanWithInputsBlocks} like this

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+						<div className={`${containerWithInputs}`}>
+							<div className={`${inputBlock}`}>
+								<span className={`${spanWithInputsBlocks}`}>Phone number</span>
+								<ValidateTextField
+									className={`${textFieldStyle}`}
+									withErrorMessage={false}	
+									placeholder={"Input your phone"}
+								/>
+							</div>
+							<div className={`${inputBlock}`}>
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: same here
PS: fix it in all places

File: StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+								/>
+							</div>
+						</div>
+						<div className={`${containerWithInputs}`}>
+							<div className={`${inputBlock}`}>
+								<span className={`${spanWithInputsBlocks}`}>Phone number</span>
+								<ValidateTextField
+									className={`${textFieldStyle}`}
+									withErrorMessage={false}	
+									placeholder={"Input your phone"}
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/index.jsx]: no need to use curly brackets

File: StudCity.Frontend/src/utils/constants.js [hunk 12]
```diff
@@ -1,5 +1,12 @@
 export const DEFAULT_AVATAR_URL = "/images/defaultAvatar.png";
-
+export const DEFAULT_BACKGROUND_URL = "/images/ChnuBackground.png"
+export const SELECT_BACKGROUND_URLS = [
+	"/images/background1.jpeg",
+	"/images/FlagUkraine.jpeg",
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/utils/constants.js]: remove duplicate values

**Review by Serhiy-dotcom — CHANGES_REQUESTED** [review]
fix this and similar issues in your files

File: StudCity.Frontend/src/UI/CustomDialog/index.jsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+	return (
+		<div className={`absolute z-50 w-full h-screen bg-[rgba(0,0,0,0.5)] ${className}`}>
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/UI/CustomDialog/index.jsx]: Pls change this color type to HEX

File: StudCity.Frontend/src/pages/RegistrationComplete/steps/UploadAvatar.jsx (hunk 1 above)
Comment (maryanchuk04) [reply StudCity.Frontend/src/pages/RegistrationComplete/steps/UploadAvatar.jsx]: I think we don`t need that useEffect

File: StudCity.Frontend/src/pages/RegistrationComplete/steps/UploadAvatar.jsx [hunk 14]
```diff
@@ -55,13 +57,15 @@ function UploadAvatar({ avatar, setAvatar }) {
 	}
 
 	return (
-		<div className="flex w-full h-full flex-col">
-			<h1 className="text-4xl text-center">Your Personal Information</h1>
+		<div className={`flex w-full h-full flex-col ${className} `}>
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/pages/RegistrationComplete/steps/UploadAvatar.jsx]: space in the end of line

File: StudCity.Frontend/src/UI/fields/UploadImageInput.jsx [hunk 15]
```diff
@@ -28,9 +28,9 @@ function UploadImageInput({ onImageSelected }) {
 				className="hidden"
 			/>
 			<Button
-				className="mx-auto w-40"
+				className={`mx-auto w-40 ${buttonStyle}`}
 				onClick={onChooseImg}>
-				Choose Image
+				{children}
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/UI/fields/UploadImageInput.jsx:33]: I think we must add buttonLabel props. No need to use children here.

**Review by maryanchuk04 — CHANGES_REQUESTED** [review]

**Review by Serhiy-dotcom — APPROVED** [review]
kudos!

**Review by maryanchuk04 — APPROVED** [review]
Good job)

**Pull request merged**